spec/snippet-loading-spec.js

```diff
@@ -0,0 +1,311 @@
+    "prefix": "foo"
+    "body": "bar1"\
+`
+      );
+      activateSnippetsPackage();
+    });
+
+    it("loads the snippets from that file", () => {
+      let snippet = null;
+
+      waitsFor(() => snippet = snippetsService.snippetsForScopes(['.foo'])['foo']);
+
+      runs(() => {
+        expect(snippet.name).toBe('foo snippet');
+        expect(snippet.prefix).toBe("foo");
+        expect(snippet.body).toBe("bar1");
+      });
+    });
+
+    describe("when that file changes", () => {
+      it("reloads the snippets", () => {
+        fs.writeFileSync(path.join(configDirPath, 'snippets.cson'), `\
+".foo":
+  "foo snippet":
+    "prefix": "foo"
+    "body": "bar2"\
+`
+        );
+
+        waitsFor("snippets to be changed", () => {
+          const snippet = snippetsService.snippetsForScopes(['.foo'])['foo'];
+          return snippet && snippet.body === 'bar2';
+        });
+
+        runs(() => {
+          fs.writeFileSync(path.join(configDirPath, 'snippets.cson'), "");
+        });
+
+        waitsFor("snippets to be removed", () => {
+          const snippet = snippetsService.snippetsForScopes(['.foo'])['foo'];
+          return snippet == null;
+        });
+      });
+    });
+  });
+
+  it("notifies the user when the user snippets file cannot be loaded", () => {
+    fs.writeFileSync(path.join(configDirPath, 'snippets.cson'), '".junk":::');
+
+    activateSnippetsPackage();
+
+    runs(() => {
+      expect(console.warn).toHaveBeenCalled();
+      if (atom.notifications != null) {
+        expect(atom.notifications.addError).toHaveBeenCalled();
+      }
+    });
+  });
+
+  describe("packages-with-snippets-disabled feature", () => {
+    it("disables no snippets if the config option is empty", () => {
+      const originalConfig = atom.config.get('core.packagesWithSnippetsDisabled');
+      atom.config.set('core.packagesWithSnippetsDisabled', []);
+
+      activateSnippetsPackage();
+      runs(() => {
+        const snippets = snippetsService.snippetsForScopes(['.package-with-snippets-unique-scope']);
+        expect(Object.keys(snippets).length).toBe(1);
+        atom.config.set('core.packagesWithSnippetsDisabled', originalConfig);
+      });
+    });
+
+    it("still includes a disabled package's snippets in the list of unparsed snippets", () => {
+      let originalConfig = atom.config.get('core.packagesWithSnippetsDisabled');
+      atom.config.set('core.packagesWithSnippetsDisabled', []);
+
+      activateSnippetsPackage();
+      runs(() => {
+        atom.config.set('core.packagesWithSnippetsDisabled', ['package-with-snippets']);
+        const allSnippets = snippetsService.getUnparsedSnippets();
+        const scopedSnippet = allSnippets.find(s => s.selectorString === '.package-with-snippets-unique-scope');
+        expect(scopedSnippet).not.toBe(undefined);
+
+        //TODO: This doesn't look right. originalConfig should be restored
+        originalConfig = atom.config.get('core.packagesWithSnippetsDisabled');
```

Comment:
My linter picked up an unused variable. Based on the others, it should be `atom.config.set('core.packagesWithSnippetsDisabled', originalConfig);` instead

Reply:
Yeah I do agree, this doesn't look right, the setting has been manipulated and we should reset it to the originalConfig, are you running into any issues applying the solution in the comment?

Reply:
No issue, I just wanted to make sure I should fix it here or a separate PR

Reply:
Normally I would fix it first and then decaf, but it's a big job and the change is simple and obvious and CI is passing, so I think that's alright for now, thanks for the heads up 👍